src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1845,29 +1845,81 @@ private FlagsExpression<FileAttributes> FormatAttributeSwitchParameters()
-        public static string Mode(PSObject instance)
+        public static string Mode(PSObject instance) => Mode(instance, excludeHardLink: false);
+
+        /// <summary>
+        /// Provides a ModeWithoutHardLink property for FileSystemInfo, without HardLinks for performance reasons.
+        /// </summary>
+        /// <param name="instance">Instance of PSObject wrapping a FileSystemInfo.</param>
+        /// <returns>A string representation of the FileAttributes, with one letter per attribute.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods")]
+        public static string ModeWithoutHardLink(PSObject instance) => Mode(instance, excludeHardLink: true);
+
+        private static string Mode(PSObject instance, bool excludeHardLink)
         {
-            if (instance == null)
-                return string.Empty;
+            string ToModeString(FileSystemInfo fileSystemInfo)
+            {
+                FileAttributes fileAttributes = fileSystemInfo.Attributes;
+                bool isReparsePoint = InternalSymbolicLinkLinkCodeMethods.IsReparsePoint(fileSystemInfo);
+                bool isDirectory = fileAttributes.HasFlag(FileAttributes.Directory);
+                bool isHardLink = isReparsePoint || excludeHardLink ? false : InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileSystemInfo);
 
-            FileSystemInfo fileInfo = (FileSystemInfo)instance.BaseObject;
-            if (fileInfo == null)
-                return string.Empty;
+                var mode = new char[]
+                {
+                    isReparsePoint || isHardLink ? 'l' : isDirectory ? 'd' : '-',
+                    fileAttributes.HasFlag(FileAttributes.Archive)   ? 'a' : '-',
+                    fileAttributes.HasFlag(FileAttributes.ReadOnly)  ? 'r' : '-',
+                    fileAttributes.HasFlag(FileAttributes.Hidden)    ? 'h' : '-',
+                    fileAttributes.HasFlag(FileAttributes.System)    ? 's' : '-',
+                };
+                return new string(mode);
+            }
+
+            return instance?.BaseObject is FileSystemInfo fileInfo
+                ? ToModeString(fileInfo)
+                : string.Empty;
+        }
+
+        /// <summary>
+        /// Provides a NameString property for FileSystemInfo.
+        /// </summary>
+        /// <param name="instance">Instance of PSObject wrapping a FileSystemInfo.</param>
+        /// <returns>Name if a file or directory, Name -> Target if symlink.</returns>
+        public static string NameString(PSObject instance)
+        {
+            return instance?.BaseObject is FileSystemInfo fileInfo
+                ? InternalSymbolicLinkLinkCodeMethods.IsReparsePoint(fileInfo)
+                    ? $"{fileInfo.Name} -> {InternalSymbolicLinkLinkCodeMethods.GetTarget(instance).First()}"
```

Comment:
Please use explicit formatting with explicit Culture.
We have an Issue to review using Culture in all code base and this help us.